test/zdtm.py

```diff
@@ -1233,8 +1233,7 @@ def __criu_act(self, action, opts=[], log=None, nowait=False):
                     return
             rst_succeeded = os.access(
                 os.path.join(__ddir, "restore-succeeded"), os.F_OK)
-            if self.__test.blocking() or (self.__sat and action == 'restore' and
-                                          rst_succeeded):
+            if (self.__test.blocking() and ret > 0) or (self.__sat and action == 'restore' and rst_succeeded):
```

Comment:
Looks correct. The only question is the case when `self.__criu == criu_rpc`, the `run` static method in this case can return negative value for any error, not related to SEGFAULT / SIGBUS. I can suggest to add a new static method (name it like `signal_exit`?) to `criu_rpc` and `criu_cli` to detect "signal exit" case. You can implement this method for `criu_cli` properly and for `criu_rpc` just always return `False` (and mark as `TODO`) for now.

Reply:
Done. Thanks for the suggestion!

Reply:
my idea was to move `ret > 0` check into `signal_exit` method. So, `if (self.__test.blocking() and ret > 0) or (self.__sat and action == 'restore' and rst_succeeded):` should be like `if (self.__test.blocking() and !self.__criu.signal_exit(ret)) or (self.__sat and action == 'restore' and rst_succeeded):`.

```
class criu_cli:
...
    @staticmethod
    def signal_exit(ret):
        return ret < 0
```

Reason is to keep all class-specific details in the class itself.